game.py

```diff
@@ -1,41 +1,82 @@
-from random import randint
+from operator import itemgetter
+
+from random import choice, randint
 
 from exceptions import NimException
 
 
 class Bot():
+    subclasses = []
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        Bot.subclasses.append(cls)
+
+    def valid_pile_indexes(self, state):
+        return [i for i, nb_stones in enumerate(state) if nb_stones >= 1]
+
     def move(self, state):
-        raise NotImplementedError
+        if not self.valid_pile_indexes(state):
+            raise NimException(f'No valid moves on state: {state}')
+        return self.move_strategy(state)
 
 
 class Random(Bot):
     """
-    Cory's bot :)
+    Bot that always chooses a random number of stones from a random pile
     """
-    pass
+    def move_strategy(self, state):
+        pile_index = choice(self.valid_pile_indexes(state))
+        stones = randint(1, state[pile_index])
+        return {'pile': pile_index, 'stones': stones}
 
 
 class Perfect(Bot):
     """
     Hunter's bot :)
+    Temporary strategy until implemented
     """
-    pass
+    def move_strategy(self, state):
+        pile_index = choice(self.valid_pile_indexes(state))
+        return {'pile': pile_index, 'stones': 1}
 
 
 class AI(Bot):
-    pass
+    """
+    Temporary strategy until implemented
+    """
+    def move_strategy(self, state):
+        pile_index = choice(self.valid_pile_indexes(state))
+        return {'pile': pile_index, 'stones': 1}
+
+
+class Greedy(Bot):
+    """
+    Bot that always takes as many stones as possible
+    """
+    def move_strategy(self, state):
+        pile_index, stones = max(enumerate(state), key=itemgetter(1))
```

Comment:
Why are we using 'itemgetter' here?

Reply:
Because otherwise it wouldn't get the correct value.
```
In [1]: state = [8, 3, 6]

In [2]: max(enumerate(state))
Out[2]: (2, 6)

In [3]: max(enumerate(state), key=itemgetter(1))
Out[3]: (0, 8)
```